services/src/main/scala/cromwell/services/auth/ecm/EcmService.scala

```diff
@@ -0,0 +1,59 @@
+package cromwell.services.auth.ecm
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers.RawHeader
+import akka.util.ByteString
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Success, Try}
+import spray.json._
+
+class EcmService(baseEcmUrl: String) {
+  private val getGithubAccessTokenApiPath = "api/oauth/v1/github/access-token"
+
+  /*
+     ECM doesn't have a standard error response format. Some of the responses contains HTML tags in it. This helper
```

Comment:
I think technically ECM always returns json format, but it's possible for things in the network layer between you and ECM (like the apache proxies, etc) to return html pages

Reply:
That's true. I have updated the comment. I also looked at the error handling in ECM and it seems it should return a standard JSON response in format of ErrorReport in all other status codes. Hence I have updated the helper method to
- only return custom error message in case its 401
- try and parse response body as JSON and extract the value associated with `message` key if found
- return the response body in case parsing failed or key `message` was not found